apps/sms/js/thread_ui.js

```diff
@@ -1113,17 +1083,32 @@ var ThreadUI = {
    * @returns {Promise} Promise that is resolved once navigation is completed.
    */
   navigateToComposer: function(parameters) {
+    var draftDiscardPromise;
+
+    // TODO: Here we should check if draft is not saved instead
     if (Compose.isEmpty()) {
-      return Navigation.toPanel('composer', parameters);
-    }
+      draftDiscardPromise = Promise.resolve();
+    } else {
+      draftDiscardPromise = Utils.confirm(
+        'unsent-message-text',
+        'unsent-message-title',
+        { text: 'unsent-message-option-discard', className: 'danger' }
+      ).then(() => this.discardDraft());
+    }
+
+    return draftDiscardPromise.then(() => {
+      // Now we'll try to find existing thread for the new message, otherwise
+      // let's fallback to new message composer.
+      var threadPromise;
```

Comment:
maybe if you initialize this here with Promise.reject() you can make the code below a little more readable.

With a comment explaining that `findThreadFromNumber` is returning a rejected Promise if it finds nothing